alloy-core/src/main/java/org/starchartlabs/alloy/core/collections/PageProvider.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2019 StarChart-Labs@github.com Authors
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+package org.starchartlabs.alloy.core.collections;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.stream.Stream;
+
+/**
+ * Represents a specialized iterator for handling elements read in distinct chunks ("pages"). Instances of
+ * {@link PageProvider} are primarily used with {@link MoreSpliterators#ofPaged(PageProvider)} to allow treated a
+ * sequence of pages as a standard Java {@link Stream}
+ *
+ * <p>
+ * Implementations of {@link PageProvider} handle underlying traversal and advancement through a sequence of pages
+ *
+ * @author romeara
+ *
+ * @param <T>
+ *            Type representing a single element of the paged data set
+ * @since 0.4.0
+ * @see MoreSpliterators#ofPaged(PageProvider)
+ */
```

Comment:
Not sure if this name is good, or something like `PageIterator`. Thoughts @nickavv ?

Reply:
yeah given that it is an Iterator I like `PageIterator`

Reply:
Will rename, sounds more reasonable to me!